raft.go

```diff
@@ -571,7 +571,12 @@ func (r *raft) maybeSendAppend(to uint64, sendIfEmpty bool) bool {
 		return false
 	}
 
-	term, errt := r.raftLog.term(pr.Next - 1)
+	index := pr.Next - 1
```

Comment:
I like the change to stop using `pr.Next` multiple times, because it's not clear when it changes or what derived indexes from the value mean (e.g. why `pr.Next - 1`?).

Should we switch the call to `raftLog.entries` to use `index + 1`? Or better, should we split `index` into `lastIndex := pr.Next - 1` and `nextIndex := pr.Next` and replace both uses of `pr.Next` (`raftLog.entries`, `UpdateOnEntriesSend`) below? The use of both indexes looks like a bug today. I'd imagine named variables would help readers understand this better.

Reply:
Done.